### DISCUSSION TITLE
automatic extension substitution

### DISCUSSION THREAD

**Discussion by jmls** [hairyhenderson/gomplate#1925] [Q&A]
we would like to keep our templates with an extension of `.tmpl` 

so , for example, we would have 

> foo.bar..tmpl

I know that you can specify a template to transform one extension to another, but that involves adding code ;) 

It would be most useful if gomplate could be configured to have a default template extension (default of .tmpl ) and if it processes a template with that extension to automatically string the ".tmpl" off the end

**Comment by hairyhenderson** [comment]
@jmls can you clarify with an example of what you're looking for?

Does this relate to the `output-map` option, perhaps? https://docs.gomplate.ca/usage/#--output-map
